src/components/CytoscapeGraph/Layout/BoxLayout.ts

```diff
@@ -130,6 +130,10 @@ export default class BoxLayout {
     this.runAsync();
   }
 
+  emit(events) {
+    this.emit(events);
```

Comment:
Do you need this?
I have the idea that this is calling itself. But I'm  not sure how cytoscape extensions work. May be it's being overridden somewhere?

Reply:
You're right, that's confusing and I need to clean it up.  I do need the `emit()` because I need to emit the `layoutstop` event when BoxLayout finally finishes, and that will allow the rest of the graph processing to then continue.  But this is just a contractual stub, the body isn't called, and that is why we don't get into a loop.

BoxLayout is a Cy layout that we have created (we register it for use in CytoscapeReactWrapper).   It means that Cytoscape will invoke it without question as a Layout (sort of like an Interface) and it is expected to support all of the necessary entry points.  Because we are the only consumer, our layout doesn't actually support everything a public layout would support, in fact up until now we only implemented the `run` entry point.  Now we have added the `emit`. 
The promise in processGraphUpdate waits on that event to be emitted.

I am going to make a commit to clean this up...  Thanks for pointing it out.

Reply:
All good.